src/result.rs

```diff
@@ -69,64 +73,83 @@ pub enum CqlType {
 
 #[napi]
 impl QueryResultWrapper {
-    pub fn from_query(
-        internal: QueryResult,
-    ) -> Result<QueryResultWrapper, IntoLegacyQueryResultError> {
-        Ok(QueryResultWrapper {
-            internal: internal.into_legacy_result()?,
-        })
+    pub fn from_query(internal: QueryResult) -> QueryResultWrapper {
+        let value = match internal.clone().into_rows_result() {
+            Ok(v) => QueryResultVariant::RowsResult(v),
+            Err(_) => QueryResultVariant::EmptyResult(internal),
+        };
```

Comment:
This is not about the number of rows returned, but about the kind of the Result response: Rows or not Rows. For non-Rows Result kinds, this conversion fails.